lineapy/plugins/pipeline_writers.py

```diff
@@ -139,107 +136,115 @@ class AirflowPipelineWriter(BasePipelineWriter):
-                        nc.safename
-                        for nc in session_artifacts.artifact_nodecollections
-                    ]
-                    task_definitions += [
-                        get_task_definition(nc, self.pipeline_name)
-                        for nc in session_artifacts.artifact_nodecollections
-                    ]
-                dependencies = {
-                    task_functions[i + 1]: {task_functions[i]}
-                    for i in range(len(task_functions) - 1)
-                }
-                task_graph = TaskGraph(
-                    nodes=task_functions,
-                    mapping={f: f for f in task_functions},
-                    edges=dependencies,
-                )
-                full_code = DAG_TEMPLATE.render(
-                    DAG_NAME=self.pipeline_name,
-                    MODULE_NAME=self.pipeline_name + "_module",
-                    OWNER=self.dag_config.get("owner", "airflow"),
-                    RETRIES=self.dag_config.get("retries", 2),
-                    START_DATE=self.dag_config.get(
-                        "start_date", "days_ago(1)"
-                    ),
-                    SCHEDULE_INTERVAL=self.dag_config.get(
-                        "schedule_interval", "*/15 * * * *"
-                    ),
-                    MAX_ACTIVE_RUNS=self.dag_config.get("max_active_runs", 1),
-                    CATCHUP=self.dag_config.get("catchup", "False"),
-                    task_definitions=task_definitions,
-                    tasks=task_functions,
-                    task_dependencies=task_graph.get_airflow_dependencies(
-                        setup_task="setup", teardown_task="teardown"
-                    ),
-                )
-
-            # Write out file
-            full_code = prettify(full_code)
-            file = self.output_dir / f"{self.pipeline_name}_dag.py"
-            file.write_text(prettify(full_code))
-            logger.info("Generated DAG file %s", file)
-        else:
+
+        # Check if the given DAG flavor is a supported/valid one
+        if dag_flavor not in AirflowDagFlavor.__members__:
             raise ValueError(
                 f'"{dag_flavor}" is an invalid airflow dag flavor.'
             )
 
+        # Construct DAG text for the given flavor
+        if (
+            AirflowDagFlavor[dag_flavor]
+            == AirflowDagFlavor.PythonOperatorPerSession
+        ):
+            full_code = self._write_operator_per_session()
+        elif (
+            AirflowDagFlavor[dag_flavor]
+            == AirflowDagFlavor.PythonOperatorPerArtifact
+        ):
+            full_code = self._write_operator_per_artifact()
+
+        # Write out file
+        full_code = prettify(full_code)
+        file = self.output_dir / f"{self.pipeline_name}_dag.py"
+        file.write_text(prettify(full_code))
+        logger.info(f"Generated DAG file: {file}")
+
+    def _write_operator_per_session(self) -> str:
```

Comment:
Lets add comments on each of these `_write_operator_per_session` and `_write_operator_per_artifact` clarifying the distinction between the two flavors. A small example in each code comment to really illustrate the difference between an operator per session vs operator per artifact would be great, but might be too long. Your call.